### PR TITLE
[receiver/gitproviderreceiver] rename metrics to match new conventions

### PR DESCRIPTION
**Pull request by adrielp**
**Description:**
Updated metrics and attributes to match newest semantic conventions for [VCSs added in the registry](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/vcs.md). 

  Attribute Changes:
    - `branch.name` is now `ref.name`
    - Added a `ref.type` attribute
    - `pull_request.state` is now `change.state`
    - `git.vendor.name` is now `vcs.vendor.name`
  Metric Changes:
    - `git.repository.count` is now `vcs.repository.count`
    - `git.repository.contributor.count` is now `vcs.repository.contributor.count`
    - `git.repository.branch.count` is now `vcs.repository.ref.count`
    - `git.repository.branch.time` is now `vcs.repository.ref.time`
    - `git.repository.branch.commit.aheadby.count` is now `vcs.repository.ref.revisions_ahead`
    - `git.repository.branch.commit.behindby.count` is now `vcs.repository.ref.revisions_behind`
    - `git.repository.branch.line.addition.count` is now `vcs.repository.ref.lines_added`
    - `git.repository.branch.line.deletion.count` is now `vcs.repository.ref.lines_deleted`
    - `git.repository.pull_request.time_open` is now `vcs.change.time_open`
    - `git.repository.pull_request.time_to_merge` is now `vcs.change.time_to_merge`
    - `git.repository.pull_request.time_to_approval` is now `vcs.change.time_to_approval`
    - `git.repository.pull_request.count` is now `vcs.change.count`

### CODE REVIEW

File: receiver/gitproviderreceiver/README.md [hunk 1]
```diff
@@ -9,15 +9,18 @@
 | [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@adrielp](https://www.github.com/adrielp), [@andrzej-stencel](https://www.github.com/andrzej-stencel) |
 
 [development]: https://github.com/open-telemetry/opentelemetry-collector#development
-<!-- end autogenerated section --> 
+<!-- end autogenerated section -->
 
-The Git Provider receiver scrapes data from Git vendors.
+The Git Provider receiver scrapes data from Version Control Systems (VCS) that use Git as their underlying technology.
```
Comment (adrielp) [inline receiver/gitproviderreceiver/README.md:14]: @crobert-1  - my only pause on this pull request is this statement right here. I want to move this to alpha and get it into contrib right after this pull request. However, my pause is at the name of the receiver. I have no plans for supporting mercurial, or SVN. However, that doesn't meant the community won't in the future. Plus, all the calls are API based, so it's not like `git` actually matters under the hood. 

Thoughts on renaming this receiver from `gitproviderreceiver` to `vcsproviderreceiver`?

File: receiver/gitproviderreceiver/README.md [hunk 2]
```diff
@@ -50,7 +53,7 @@ receivers:
         scrapers:
             github:
                 metrics:
-                    git.repository.contributor.count:
+                    vcs.repository.contributor.count:
```
Comment (crobert-1) [inline receiver/gitproviderreceiver/README.md:56]: This rename isn't in the changelog, should it be?

File: receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go [hunk 3]
```diff
@@ -134,8 +136,8 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 				// See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/gitproviderreceiver/internal/scraper/githubscraper/README.md#github-limitations
 				// for more information as to why `BehindBy` and `AheadBy` are
```
Comment (crobert-1) [inline receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go:137]: It looks like there are some comments referring to old metric names, they should probably be updated, as well as in the githubscraper README.

File: receiver/gitproviderreceiver/metadata.yaml [hunk 4]
```diff
@@ -11,119 +11,121 @@ status:
...
 
-# this might need to be unique per sub receiver implementation
 resource_attributes:
   organization.name:
     enabled: true
-    description: Git Organization or Project Name
+    description: VCS Organization
     type: string
-  git.vendor.name:
+  vcs.vendor.name:
```
Comment (crobert-1) [inline receiver/gitproviderreceiver/metadata.yaml:19]: Thie resource attribute rename isn't in the changelog, should it be?

`git.vendor.name` -> `vcs.vendor.name`

File: receiver/gitproviderreceiver/README.md (hunk 1 above)
Comment (adrielp) [reply receiver/gitproviderreceiver/README.md:14]: Discussed async, will rename as separate pull request. I have no intentions of contributing code to support systems like mercurial, etc, however if the community wants it, they're welcome to. Despite that intention, this still more closely aligns with SemConv v1.27.0 (which is getting ready to be released).

**Review by crobert-1 — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]
Awesome!

**Pull request merged**